goth_tests/domain/ya-provider/test_provider_break_agreement.py

```diff
@@ -0,0 +1,230 @@
+"""End to end tests for requesting WASM tasks using goth REST API clients."""
+
+import logging
+from pathlib import Path
+from typing import List, Tuple
+
+import pytest
+
+from goth.address import (
+    PROXY_HOST,
+    YAGNA_REST_URL,
+)
+from goth.configuration import load_yaml, Override, Configuration
+from goth.node import node_environment
+from goth.runner import Runner
+from goth.runner.container.payment import PaymentIdPool
+from goth.runner.container.yagna import YagnaContainerConfig
+from goth.runner.probe import RequestorProbe, ProviderProbe
+
+from goth_tests.helpers.activity import run_activity, wasi_exe_script, wasi_task_package
+from goth_tests.helpers.negotiation import negotiate_agreements, DemandBuilder
+from goth_tests.helpers.payment import pay_all
+
+logger = logging.getLogger("goth.test.breaking-agreement")
+
+
+def build_demand(
+    requestor: RequestorProbe,
+    runner: Runner,
+    task_package_template: str,
+    require_debit_notes=True,
+):
+    """Simplifies creating demand."""
+
+    task_package = task_package_template.format(
+        web_server_addr=runner.host_address, web_server_port=runner.web_server_port
+    )
+
+    demand = (
+        DemandBuilder(requestor)
+        .props_from_template(task_package)
+        .property("golem.srv.caps.multi-activity", True)
+        .constraints(
+            "(&(golem.com.pricing.model=linear)\
+            (golem.srv.caps.multi-activity=true)\
+            (golem.runtime.name=wasmtime))"
+        )
+    )
+
+    if require_debit_notes:
+        demand = demand.property("golem.com.payment.debit-notes.accept-timeout?", 8)
+    return demand.build()
+
+
+def _create_runner(
+    common_assets: Path, config_overrides: List[Override], log_dir: Path
+) -> Tuple[Runner, Configuration]:
+    goth_config = load_yaml(
+        common_assets / ".." / "domain" / "ya-provider" / "assets" / "goth-config.yml",
```

Comment:
Please take a look at the comment regarding the same function in `test_provider_multi_activity.py`.